src/operator/tensor/matrix_op-inl.h

```diff
@@ -601,18 +590,127 @@ void SliceCsrImpl(const SliceParam &param, const OpContext& ctx,
   });
 }
 
+// slice a CSRNDArray for two dimensions
+struct SliceDimTwoCsrAssign {
+  template<typename IType, typename RType, typename DType>
+  MSHADOW_XINLINE static void Map(int i, IType* out_idx, DType* out_data,
+                                  const RType* out_indptr, const IType* in_idx,
+                                  const DType* in_data, const RType* in_indptr,
+                                  const int begin, const int end) {
+    RType ind = out_indptr[i];
+    for (RType j = in_indptr[i]; j < in_indptr[i+1]; j++) {
+      if (in_idx[j] >= end) {
```

Comment:
Any reason  to not just do  if (in_idx[j] >= begin_col && in_idx < end_col)

Reply:
Indices of csr ndarray is in ascending order per row. So if `indice > end`, there is no need to continue the loop.

Reply:
I was suggesting this change for readability. Also, you would be doing the checks for all in_idx[j] < begin_col which will be avoided with the change.

Reply:
I used this condition, in_idx[j] >= begin_col && in_idx < end_col, at the first time. But according to @eric-haibin-lin 's comments, this logic should be changed to a if/else logic which can jump out of the loop since indices are sorted per row.